kong/utils.go

```diff
@@ -741,3 +741,119 @@ func FillPluginsDefaults(plugin *Plugin, schema Schema) error {
 func FillPluginsDefaultsWithOpts(plugin *Plugin, schema map[string]interface{}, opts FillRecordOptions) error {
 	return fillConfigRecordDefaultsAutoFields(plugin, schema, opts)
 }
+
+func deleteAndCollapseMap(config map[string]interface{}, path []string) {
+	key := path[0]
+	if len(path) == 1 {
+		delete(config, key)
+	} else {
+		if nested, ok := config[key].(map[string]interface{}); ok {
+			deleteAndCollapseMap(nested, path[1:])
+			if len(nested) == 0 {
```

Comment:
Out of curiosity, when can this case arise?

Reply:
I wanted to cover a scenario when ultimately the entire nested map needs to collapse. Like lets say we have a structure:
```json
{
  "a": {
    "a1": 1,
    "a2": 2,
    "a3": 3
  },
  "b": {
    "b1": 1,
    "b2": 2,
    "b3": 3,
  }   
}
```

and let's say we have deprecated: `a.a1`, `a.a2`, `a.a3` and `b.b1`, `b.b2`. When collapsing this I'd like to end up with following JSON:

```json
{
  "b": {
    "b3": 3
  }
}
```